source/core/Events.cpp

```diff
@@ -155,6 +169,10 @@ class EventOracle {
     void DoneProcessingEvent(EventQueueID qID) {
         return _qObject[qID].DoneProcessingEvent();
     }
+    void ClearEventQueue(EventQueueID qID) {
+        if (qID < _qObject.size())
```

Comment:
Why not assert qID < _qObject.size() and then call _qObject[qID].ClearQueue();?

Reply:
This is called by clean up procs, so the queue could have been already freed explicitly, so this is not an 'assert' type condition, it could really be out of range for legitimate reasons.